miio/device.py

```diff
@@ -252,31 +284,21 @@ def actions(self) -> Dict[str, ActionDescriptor]:
 
     def settings(self) -> Dict[str, SettingDescriptor]:
         """Return device settings."""
-        settings = self.status().settings()
-        for setting in settings.values():
-            # TODO: Bind setter methods, this should probably done only once during init.
-            if setting.setter is None:
-                # TODO: this is ugly, how to fix the issue where setter_name is optional and thus not acceptable for getattr?
-                if setting.setter_name is None:
-                    raise Exception(
-                        f"Neither setter or setter_name was defined for {setting}"
-                    )
-
-                setting.setter = getattr(self, setting.setter_name)
-            if (
-                isinstance(setting, EnumSettingDescriptor)
-                and setting.choices_attribute is not None
-            ):
-                retrieve_choices_function = getattr(self, setting.choices_attribute)
-                setting.choices = retrieve_choices_function()  # This can do IO
+        if self._settings is None:
+            raise Exception(
+                "setting descriptors schould first be cached using the info() call"
```

Comment:
```suggestion
                "setting descriptors should first be cached using the info() call"
```
Btw, you often type `schould` instead of `should` :-)